src/sync/app_credentials.hpp

```diff
@@ -78,6 +93,16 @@ struct AppCredentials {
     // Construct and return credentials from a username and password.
     static AppCredentials username_password(std::string username, std::string password);
 
+    // Construct and return credentials with the payload.
+    static AppCredentials function(std::string payload_json);
```

Comment:
What is the format of this string? An JSON array? Something else? Maybe add it to the description?

Reply:
its going to be a mongodb document, I'll update the docs to reflect it